### DISCUSSION TITLE
Some Image Tags are Broken

### DISCUSSION THREAD

**Discussion by davefilip** [weather-gov/api#694] [General]
All,

I an frequently having problems with some of the icons returned from:

    https://api.weather.gov/gridpoints/ALY/92,50/forecast/hourly

For example, in the following response:

    "periods": [
            {
                "number": 1,
                "name": "",
                "startTime": "2024-01-01T23:00:00-05:00",
                "endTime": "2024-01-02T00:00:00-05:00",
                "isDaytime": false,
                "temperature": 23,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "probabilityOfPrecipitation": {
                    "unitCode": "wmoUnit:percent",
                    "value": 0
                },
                "dewpoint": {
                    "unitCode": "wmoUnit:degC",
                    "value": -7.7777777777777777
                },
                "relativeHumidity": {
                    "unitCode": "wmoUnit:percent",
                    "value": 81
                },
                "windSpeed": "6 mph",
                "windDirection": "NW",
                "icon": "https://api.weather.gov/icons/land/night/sct,0?size=small",
                "shortForecast": "Partly Cloudy",
                "detailedForecast": ""
            },

it returns the following icon:

    https://api.weather.gov/icons/land/night/sct,0?size=small

However, attempting to retrieve the icon returns an error:

  {
    "correlationId": "13f7665e",
    "title": "Bad Request",
    "type": "https://api.weather.gov/problems/BadRequest",
    "status": 400,
    "detail": "Bad Request",
    "instance": "https://api.weather.gov/requests/13f7665e"
  }

It seems like if I remove the ',0' it sometimes (usually?) works, as in:

    https://api.weather.gov/icons/land/night/sct?size=small

but before I put in code to automatically do that, I would like to understand what the problem is, and if it is always consistent (e.g., should I always change /sct,0/ to /sct/) ?

Please advise.

Thanks,

Dave Filip
dfilip@colornetlabs.com

**Comment by chaunceygardiner** [comment]
It’s an old bug.  See this from April 2023:
https://github.com/weather-gov/api/discussions/633

I responded on that discussion that I replace “,0?” with “?”.  Note: I believe this problem only shows up for the hourly forecast.  For the 12-hour, zero percent POP is not appended to the URL.

I don’t recall anyone saying this may be fixed one day.

**Reply by davefilip** [reply]
Great, thanks for confirming the issue.  Hopefully it will be fixed at some point, but at least until then I understand what is causing this.  To be honest, until looking closer at this, I didn't realize that the ",#' meant that the icons were being dynamically altered based on the percent precipitation, but now that I look closer at it, I get that.

I'm just happy that on Jan 1, 2024 the /forecast/hourly endpoint started working again!  Since it broke towards the end of 2023, I was afraid I was going to have to find a new Weather API.
